src/data/finsupp/basic.lean

```diff
@@ -774,8 +775,76 @@ end
 @[simp] lemma graph_eq_empty {f : α →₀ M} : f.graph = ∅ ↔ f = 0 :=
 (graph_injective α M).eq_iff' graph_zero
 
+/-- When turned into a list of key/value pairs, a graph has no duplicate keys. -/
+lemma graph_nodupkeys (f : α →₀ M) : (list.map prod.to_sigma f.graph.to_list).nodupkeys :=
+begin
+  rw [list.nodupkeys, list.keys, list.map_map, prod.fst_to_sigma_comp, list.nodup_map_iff_inj_on],
+  { rintros ⟨b, m⟩ hb ⟨c, n⟩ hc (rfl : b = c),
+    rw [mem_to_list, finsupp.mem_graph_iff] at hb hc,
+    dsimp at hb hc,
+    rw [←hc.1, hb.1] },
+  { apply nodup_to_list }
+end
+
 end graph
 
+end finsupp
+
+/-! ### Declarations about `list.lookup_finsupp` and `alist.lookup_finsupp` -/
+
+namespace list
+variable [has_zero M]
+
+/-- Converts a list of key/value pairs into a finitely supported function via `list.lookup`, sending
+absent keys to zero. -/
+def lookup_finsupp (l : list (sigma (λ x : α, M))) : α →₀ M :=
+{ support := l.keys.to_finset.filter $ λ x, (0 : M) ∉ l.lookup x,
```

Comment:
```suggestion
{ support := ⟨(l.dedup_keys.filter (λ x, x.2 ≠ 0)).keys, sorry⟩,
```

Reply:
I prefer the current definition for the support: it's more direct, which makes proving `mem_support_to_fun` much easier. If we ever need it, the equivalence with your support should follow from `list.lookup_dedupkeys`.

Reply:
I think my version is computationally faster; yours has quadratic complexity, mine is linear.